src/nunit.analyzers/SameActualExpectedValue/SameActualExpectedValueAnalyzer.cs

```diff
@@ -38,7 +38,7 @@ public class SameActualExpectedValueAnalyzer : BaseAssertionAnalyzer
             {
                 context.ReportDiagnostic(Diagnostic.Create(
                     descriptor,
-                    expected.GetLocation(),
+                    expected!.GetLocation(),
```

Comment:
In this case we know for sure that `expected` is not null because of line 35:
`.Where(e => actualExpression.IsEquivalentTo(e));`
(`IsEquivalent` returns false if input is null)

Reply:
> 
> 
> Personally, I would prefer that we don't have too many `!`s, but on the other hand it would also be annoying to have too many unnecessary null checks just to avoid `!`.

Debug.Assert or equivalent can help fill the gap too. If you were using ReferenceAssemblyAnnotator, that would be able to replace the `!`.